### PR TITLE
IGNITE-8152 Forbid empty sql schema name

### PR DESCRIPTION
**Pull request by Aleksei-Litsov**
PR (Forbid empty sql schema name) from Aleksei_Litsov@epam.com

### CODE REVIEW

**Comment by Aleksei-Litsov** [comment]
PR (Forbid empty sql schema name) from Aleksei_Litsov@epam.com

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/H2ConnectionWrapper.java [hunk 1]
```diff
@@ -73,6 +73,11 @@ public void schema(@Nullable String schema) {
     public Connection connection(@Nullable String schema) {
         if (schema != null && !F.eq(this.schema, schema)) {
             try {
+
+                if(schema != null && schema.trim().isEmpty()){
```
Comment (joooger) [inline modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/H2ConnectionWrapper.java]: here `schema` is always not null

File: modules/indexing/src/test/java/org/apache/ignite/sqltests/BaseSqlTest.java [hunk 2]
```diff
@@ -351,6 +353,10 @@ protected Result executeFrom(String qry, Ignite node) {
         return executeFrom(new SqlFieldsQuery(qry), node);
     }
 
+    protected Result executeFrom(String qry, Ignite node, String schema) {
```
Comment (AMashenkov) [inline modules/indexing/src/test/java/org/apache/ignite/sqltests/BaseSqlTest.java]: Missed javadoc

File: modules/indexing/src/test/java/org/apache/ignite/sqltests/BaseSqlTest.java [hunk 3]
```diff
@@ -1249,6 +1255,36 @@ public void testFullOuterDistributedJoinIsNotSupported() {
         });
     }
 
+    /**
+     * Check schema for validation
+     */
+
```
Comment (AMashenkov) [inline modules/indexing/src/test/java/org/apache/ignite/sqltests/BaseSqlTest.java]: Redundant empty line

File: modules/indexing/src/test/java/org/apache/ignite/sqltests/BaseSqlTest.java [hunk 4]
```diff
@@ -1249,6 +1255,36 @@ public void testFullOuterDistributedJoinIsNotSupported() {
...
+    @Rule
+    public ExpectedException expectedEx = ExpectedException.none();
+
+    @Test
```
Comment (AMashenkov) [inline modules/indexing/src/test/java/org/apache/ignite/sqltests/BaseSqlTest.java]: Missed javadoc

File: modules/indexing/src/test/java/org/apache/ignite/sqltests/BaseSqlTest.java [hunk 5]
```diff
@@ -1249,6 +1255,36 @@ public void testFullOuterDistributedJoinIsNotSupported() {
...
+    public void testCheckEmptySchema() {
+
```
Comment (AMashenkov) [inline modules/indexing/src/test/java/org/apache/ignite/sqltests/BaseSqlTest.java]: Empty line

File: modules/indexing/src/test/java/org/apache/ignite/sqltests/BaseSqlTest.java [hunk 6]
```diff
@@ -1249,6 +1255,36 @@ public void testFullOuterDistributedJoinIsNotSupported() {
...
+            executeFrom(sqlQuery, node, "");
+            executeFrom(sqlQuery, node, " ");
+            assertTrue("Check valid schema",
+                    executeFrom(sqlQuery, node, "PUBLIC").values().stream().count() > 0
+            );
+            assertTrue("Check null schema",
+                    executeFrom(sqlQuery, node, null).values().stream().count() > 0
+            );
+        });
+
```
Comment (AMashenkov) [inline modules/indexing/src/test/java/org/apache/ignite/sqltests/BaseSqlTest.java]: Empty line.

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/H2ConnectionWrapper.java (hunk 1 above)
Comment (Aleksei-Litsov) [reply modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/H2ConnectionWrapper.java]: fixed

File: modules/indexing/src/test/java/org/apache/ignite/sqltests/BaseSqlTest.java (hunk 2 above)
Comment (Aleksei-Litsov) [reply modules/indexing/src/test/java/org/apache/ignite/sqltests/BaseSqlTest.java]: fixed

File: modules/indexing/src/test/java/org/apache/ignite/sqltests/BaseSqlTest.java (hunk 3 above)
Comment (Aleksei-Litsov) [reply modules/indexing/src/test/java/org/apache/ignite/sqltests/BaseSqlTest.java]: fixed

File: modules/indexing/src/test/java/org/apache/ignite/sqltests/BaseSqlTest.java (hunk 4 above)
Comment (Aleksei-Litsov) [reply modules/indexing/src/test/java/org/apache/ignite/sqltests/BaseSqlTest.java]: fixed

File: modules/indexing/src/test/java/org/apache/ignite/sqltests/BaseSqlTest.java (hunk 5 above)
Comment (Aleksei-Litsov) [reply modules/indexing/src/test/java/org/apache/ignite/sqltests/BaseSqlTest.java]: fixed

File: modules/indexing/src/test/java/org/apache/ignite/sqltests/BaseSqlTest.java (hunk 6 above)
Comment (Aleksei-Litsov) [reply modules/indexing/src/test/java/org/apache/ignite/sqltests/BaseSqlTest.java]: fixed

**Comment by Aleksei-Litsov** [comment]
PR (Forbid empty sql schema name) from Aleksei_Litsov@epam.com

**Pull request merged**